Player/GrenadeLauncher.gd

```diff
@@ -72,78 +71,73 @@ func update_aim() -> void:
 	_throw_velocity = Vector3.UP * velocity_up + forward_velocity
 
-	# Redraw the grenade's motion preview
-#	_grenade_path.curve.clear_points()
-	_trail_points.clear()
+
+func _draw_throw_path() -> void:
 	const TIME_STEP := 0.05
-	var time_current := 0.0
-	var end_time := time_to_land + 0.5
-	while time_current < end_time:
-		var point := _throw_velocity * time_current + Vector3.DOWN * gravity * 0.5 * time_current * time_current
-		_trail_points.append(point)
-		time_current += TIME_STEP
-	_generate_mesh()
+	const TRAIL_WIDTH := 0.25
 
+	var forward_direction = Vector3(_throw_velocity.x, 0.0, _throw_velocity.z).normalized()
+	var left_direction := Vector3.UP.cross(forward_direction)
+	var offset_left = left_direction * TRAIL_WIDTH / 2.0
+	var offset_right = -left_direction * TRAIL_WIDTH / 2.0
 
-func _generate_mesh() -> void:
 	var st := SurfaceTool.new()
 	st.begin(Mesh.PRIMITIVE_TRIANGLES)
-	
-	# We'll use trail_direction to get the left and right points difference from
-	# the _trail_points 
-	var trail_direction = (_trail_points[-1] - _trail_points[0]).normalized()
-	trail_direction.y = 0.0
-	var cross_dir := Vector3.UP.cross(trail_direction)
-	var mesh_left_diff = cross_dir * TRAIL_SIZE/2.0
-	var mesh_right_diff = -cross_dir * TRAIL_SIZE/2.0
-
-	for i in range(_trail_points.size()-1):
-		# We'll create 2 triangles on each iteration, representing the quad of
-		# one section of the path (between two points of _trail_points)
-		
-		# _trail_points represents the points in the "center" of the path, so
-		# we'll derive the two points in the left and the two points in the right
-		var trail_point_left_1 = _trail_points[i] + mesh_left_diff
-		var trail_point_right_1 = _trail_points[i] + mesh_right_diff
-		var trail_point_left_2 = _trail_points[i+1] + mesh_left_diff
-		var trail_point_right_2 = _trail_points[i+1] + mesh_right_diff
+
+	var end_time := _time_to_land + 0.5
+	var point_previous = Vector3.ZERO
+	var time_current := 0.0
+	var uv_progress_factor := TIME_STEP / end_time
+	# We'll create 2 triangles on each iteration, representing the quad of one
+	# section of the path
+	while time_current < end_time:
+		time_current += TIME_STEP
+		var point_current := _throw_velocity * time_current + Vector3.DOWN * gravity * 0.5 * time_current * time_current
+
+		# Our point coordinates are at the center of the path, so we need to calculate vertices
+		var trail_point_left_1 = point_current + offset_left
+		var trail_point_right_1 = point_current + offset_right
+		var trail_point_left_2 = point_previous + offset_left
+		var trail_point_right_2 = point_previous + offset_right
 		
 		# UV position goes from 0 to 1, so we normalize the current iteration
 		# to get the progress in the UV texture
-		var uv_progress_1 = (float(i)/(_trail_points.size()-1.0))
-		var uv_progress_2 = (float(i+1.0)/(_trail_points.size()-1.0))
+		var uv_progress_1 = time_current * uv_progress_factor
+		var uv_progress_2 = uv_progress_1 - uv_progress_factor
```

Comment:
Looping just once simplifies the calculations so it also increases performance, not in a noticeable way, but it's a good practice to show: code is both simpler and faster